x/pki/handler_test.go

```diff
@@ -135,1703 +135,1983 @@ func Setup(t *testing.T) *TestSetup {
+			require.Equal(t, 1, len(allApprovedCertificates))
+			require.Equal(t, testconstants.RootSubject, allApprovedCertificates[0].Subject)
+			require.Equal(t, testconstants.RootSubjectKeyID, allApprovedCertificates[0].SubjectKeyId)
+
+			// check that no proposed certificate revocations have been created
+			allProposedCertificateRevocations, _ := queryAllProposedCertificateRevocations(setup)
+			require.NoError(t, err)
+			require.Equal(t, 0, len(allProposedCertificateRevocations))
+
+			// check that no child certificate identifiers are now registered for root certificate
+			_, err = queryChildCertificates(setup, testconstants.RootSubject, testconstants.RootSubjectKeyID)
+			require.Error(t, err)
+			require.Equal(t, codes.NotFound, status.Code(err))
+
+			// check that no child certificate identifiers are registered for revoked intermediate certificate
+			_, err = queryChildCertificates(setup, testconstants.IntermediateSubject, testconstants.IntermediateSubjectKeyID)
+			require.Error(t, err)
+			require.Equal(t, codes.NotFound, status.Code(err))
+
+			// check that no child certificate identifiers are registered for revoked leaf certificate
+			_, err = queryChildCertificates(setup, testconstants.LeafSubject, testconstants.LeafSubjectKeyID)
+			require.Error(t, err)
+			require.Equal(t, codes.NotFound, status.Code(err))
+		})
+	})
+
+	t.Run("RejectX509RootCert", func(t *testing.T) {
+
+		t.Run("TwoRejectApprovalsAreNeeded", func(t *testing.T) {
+			setup := Setup(t)
+
+			// propose x509 root certificate by account Trustee1
+			proposeAddX509RootCert, err := ProposeX509RootCert(setup, setup.Trustee1, testconstants.RootCertPem, testconstants.Info, testconstants.Vid)
+			require.NoError(t, err)
+
+			// reject x509 root certificate by account Trustee2
+			_, err = RejectX509RootCert(setup, setup.Trustee2, testconstants.RootSubject, testconstants.RootSubjectKeyID, testconstants.Info)
+			require.NoError(t, err)
+
+			// certificate should be in the entity <Proposed X509 Root Certificate>, because we haven't enough reject approvals
+			proposedCertificate, err := queryProposedCertificate(setup, testconstants.RootSubject, testconstants.RootSubjectKeyID)
+			require.NoError(t, err)
+
+			// check proposed certificate
+			require.Equal(t, proposeAddX509RootCert.Cert, proposedCertificate.PemCert)
+			require.Equal(t, proposeAddX509RootCert.Signer, proposedCertificate.Owner)
+			require.Equal(t, testconstants.RootSubject, proposedCertificate.Subject)
+			require.Equal(t, testconstants.RootSubjectKeyID, proposedCertificate.SubjectKeyId)
+			require.Equal(t, testconstants.RootSerialNumber, proposedCertificate.SerialNumber)
+			require.Equal(t, setup.Trustee1.String(), proposedCertificate.Approvals[0].Address)
+			require.Equal(t, testconstants.Info, proposedCertificate.Approvals[0].Info)
+			require.Equal(t, setup.Trustee2.String(), proposedCertificate.Rejects[0].Address)
+			require.Equal(t, testconstants.Info, proposedCertificate.Rejects[0].Info)
```

Comment:
Can we move these checks to another function?
Just an idea

Reply:
There are 5 repetitions with such checks in the test cases TwoRejectApprovalsAreNeeded and DoubleTime. I've separated them into functions in a new commit